### PR TITLE
boards: arm: Add xtools as a supported toolchain

### PR DESCRIPTION
**Pull request by galak**
Update all ARM boards to have xtools as a supported toolchain.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]

**Review by ydamigos — APPROVED** [review]

**Review by nashif — APPROVED** [review]

File: boards/arm/cc3220sf_launchxl/cc3220sf_launchxl.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@ arch: arm
 toolchain:
   - zephyr
   - gnuarmemb
+  - xtools
```
Comment (GAnthony) [inline boards/arm/cc3220sf_launchxl/cc3220sf_launchxl.yaml]: @galak, I wonder, would cc3220sf ever need to build with xtensa toolchain?

Comment (nashif) [reply boards/arm/cc3220sf_launchxl/cc3220sf_launchxl.yaml]: this is not xtensa, this is xtools

Comment (GAnthony) [reply boards/arm/cc3220sf_launchxl/cc3220sf_launchxl.yaml]: ah, OK.  MacOS.

File: boards/arm/mps2_an385/mps2_an385.yaml [hunk 2]
```diff
@@ -7,5 +7,6 @@ toolchain:
   - zephyr
   - gnuarmemb
   - xtools
+  - xtools
```
Comment (pfalcon) [inline boards/arm/mps2_an385/mps2_an385.yaml]: @galak: There was already xtools for mps2_an385.yaml, so now there're two in row.

**Pull request merged**